H4LCppModule.py

```diff
@@ -112,12 +133,30 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         self.out.branch("pTj1",  "F")
         self.out.branch("etaj1",  "F")
         self.out.branch("phij1",  "F")
-        self.out.branch("mj2",  "F")
         self.out.branch("pTj2",  "F")
         self.out.branch("etaj2",  "F")
         self.out.branch("phij2",  "F")
-
-
+        self.out.branch("mj2",  "F")
+        self.out.branch("pileupWeight",  "F")
+        self.out.branch("dataMCWeight_new",  "F")
+        self.out.branch("prefiringWeight",  "F")
+        self.out.branch("passedTrig",  "O")
+        self.out.branch("passedFullSelection",  "O")
+        self.out.branch("passedZ4lSelection",  "O")
+        self.out.branch("passedZ4lZ1LSelection",  "O")
+        self.out.branch("passedZ4lZXCRSelection",  "O")
+        self.out.branch("passedZXCRSelection",  "O")
+        self.out.branch("passedFiducialSelection",  "O")
+        GENHlepNum = 4
+        GENZNum = 2
+        self.out.branch("GENlep_MomId",  "I", lenVar = "nGENLeptons")
+        self.out.branch("GENlep_MomMomId",  "I", lenVar = "nGENLeptons")
+        self.out.branch("GENZ_MomId",  "I", lenVar = "nVECZ")
+        self.out.branch("GENZ_DaughtersId",  "I", lenVar = "GENZNum")
+        self.out.branch("GENlep_Hindex",  "I", lenVar = "GENHlepNum")
+        self.out.branch("lep_Hindex",  "I", lenVar = "GENHlepNum")
+        self.out.branch("GENlep_id",  "I", lenVar = "nGENLeptons")
+        self.out.branch("lep_genindex",  "I", lenVar = "Lepointer")
```

Comment:
HI @YujiLee301 ,

Several branches are not in the output root file. You need to add them in the `keep_and_drop.txt` file, only then they will appear.